datatypes/src/raster/grid.rs

```diff
@@ -230,6 +230,16 @@ impl GridBounds for GridShape3D {
     }
 }
 
+pub fn grid_idx_iter_2d<D>(t: &D) -> impl Iterator<Item = GridIdx2D>
```

Comment:
Please add a comment on what this does.

Reply:
it generates an iterator over all pixel indexes in a grid

Reply:
7787a35